File: scripts/diagnostic-complet.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');
require('dotenv').config({ path: '.env.local' });

const supabase = createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL,
  process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY
);

async function diagnosticComplet() {
  console.log("\n🔍 DIAGNOSTIC COMPLET DES BASES DE DONNÉES\n");
  
  const tables = [
    { name: 'exportateurs', label: '📦 Exportateurs généraux' },
    { name: 'laboratoires_pharmaceutiques', label: '💊 Laboratoires pharmaceutiques' },
    { name: 'entreprises_dechets', label: '♻️ Traitement des déchets' }
  ];
  
  let rapportGlobal = {
    date: new Date().toISOString(),
    totalGeneral: 0,
    tables: {}
  };
  
  for (const table of tables) {
    console.log(`\n${table.label}`);
    console.log("═".repeat(60));
    
    // Compter le nombre total
    const { count, error } = await supabase
      .from(table.name)
      .select('*', { count: 'exact', head: true });
    
    if (error) {
      console.log(`❌ Erreur: ${error.message}`);
      continue;
    }
    
    console.log(`📊 TOTAL: ${count} enregistrements`);
    rapportGlobal.totalGeneral += count || 0;
    
    // Récupérer les 1000 premiers pour analyse
    const { data: echantillon, error: fetchError } = await supabase
      .from(table.name)
      .select('*')
      .limit(1000);
    
    if (fetchError) {
      console.log(`❌ Erreur chargement: ${fetchError.message}`);
      continue;
    }
    
    if (!echantillon || echantillon.length === 0) {
      console.log(`⚠️ Aucune donnée dans cette table`);
      rapportGlobal.tables[table.name] = {
        total: count || 0,
        echantillon: []
      };
      continue;
    }
    
    // Analyser les colonnes disponibles
    const colonnes = Object.keys(echantillon[0] || {});
    console.log(`📋 Colonnes: ${colonnes.join(', ')}`);
    
    // Analyser les sources
    const sources = {};
    echantillon.forEach(e => {
      if (e.source) {
        sources[e.source] = (sources[e.source] || 0) + 1;
      }
    });
    
    console.log(`📌 Sources:`);
    Object.entries(sources).forEach(([src, nb]) => {
      console.log(`   - ${src}: ${nb} entrées`);
    });
    
    // Afficher les 5 premiers enregistrements
    console.log(`\n🔎 Aperçu des 5 premiers:`);
    echantillon.slice(0, 5).forEach((e, i) => {
      console.log(`\n   ${i+1}. ${e.nom || 'Sans nom'}`);
      if (e.wilaya) console.log(`      Wilaya: ${e.wilaya}`);
      if (e.secteur) console.log(`      Secteur: ${e.secteur}`);
      if (e.activite) console.log(`      Activité: ${e.activite}`);
      if (e.email) console.log(`      Email: ${e.email}`);
      if (e.telephone) console.log(`      Tél: ${e.telephone}`);
    });
    
    // Détection des doublons potentiels dans l'échantillon
    const nomsVus = new Map();
    const doublons = [];
    
    echantillon.forEach(e => {
      if (!e.nom) return;
      const nomNormalise = e.nom.toLowerCase().replace(/[^a-z0-9]/g, '');
      if (nomsVus.has(nomNormalise)) {
        doublons.push({
          original: nomsVus.get(nomNormalise),
          doublon: e.nom
        });
      } else {
        nomsVus.set(nomNormalise, e.nom);
      }
    });
    
    if (doublons.length > 0) {
      console.log(`\n⚠️ Doublons potentiels dans l'échantillon:`);
      doublons.slice(0, 10).forEach((d, i) => {
        console.log(`   ${i+1}. "${d.original}" ≈ "${d.doublon}"`);
      });
      if (doublons.length > 10) {
        console.log(`   ... et ${doublons.length - 10} autres`);
      }
    } else {
      console.log(`\n✅ Aucun doublon évident dans l'échantillon`);
    }
    
    // Sauvegarder les stats
    rapportGlobal.tables[table.name] = {
      total: count || 0,
      colonnes,
      sources,
      doublonsPotentiels: doublons.length,
      echantillon: echantillon.slice(0, 10).map(e => ({
        id: e.id,
        nom: e.nom,
        source: e.source,
        wilaya: e.wilaya,
        secteur: e.secteur,
        activite: e.activite
      }))
    };
    
    console.log("\n" + "═".repeat(60));
  }
  
  // RAPPORT FINAL
  console.log("\n" + "=".repeat(70));
  console.log("📊 RAPPORT DE DIAGNOSTIC FINAL");
  console.log("=".repeat(70));
  console.log(`📅 Date: ${new Date().toLocaleString()}`);
  console.log(`📦 TOTAL GÉNÉRAL: ${rapportGlobal.totalGeneral} enregistrements`);
  console.log("-".repeat(70));
  
  for (const [nom, stats] of Object.entries(rapportGlobal.tables)) {
    console.log(`\n${nom}:`);
    console.log(`   📊 Total: ${stats.total} enregistrements`);
    if (stats.sources) {
      console.log(`   📌 Sources: ${Object.keys(stats.sources).join(', ') || 'aucune'}`);
    }
    console.log(`   ⚠️ Doublons potentiels: ${stats.doublonsPotentiels || 0}`);
    
    if (stats.echantillon && stats.echantillon.length > 0) {
      console.log(`   🔎 Exemples:`);
      stats.echantillon.slice(0, 3).forEach(e => {
        console.log(`      - ${e.nom || 'Sans nom'}`);
      });
    }
  }
  
  console.log("\n" + "=".repeat(70));
  
  // Sauvegarder le rapport complet
  fs.writeFileSync(
    'diagnostic_complet.json',
    JSON.stringify(rapportGlobal, null, 2),
    'utf8'
  );
  
  console.log(`\n✅ Rapport sauvegardé dans: diagnostic_complet.json`);
  console.log("\n✅ DIAGNOSTIC TERMINÉ!");
}

// Exécution
diagnosticComplet().catch(error => {
  console.error("❌ ERREUR FATALE:", error);
});

File: scripts/diagnostic_complet.py
# scripts/diagnostic_complet.py
import os
import csv
import json
from supabase import create_client
from dotenv import load_dotenv
from datetime import datetime

print("="*80)
print("🔬 DIAGNOSTIC COMPLET - NIVEAU EXPERT")
print("="*80)

# 1. CONNEXION
load_dotenv('.env.local')
url = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")
supabase = create_client(url, key)

print(f"\n📅 Date: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
print("="*80)

# 2. ANALYSE DES TABLES EXISTANTES
print("\n📋 1. TABLES EXISTANTES DANS SUPABASE")
print("-"*60)

tables_testees = [
    'wilayas', 'sectors', 'legal_statuses', 'company_types',
    'exporters', 'companies', 'profiles', 'users'
]

tables_existantes = []

for table in tables_testees:
    try:
        result = supabase.table(table).select('*', count='exact').limit(1).execute()
        print(f"✅ {table}: {result.count} enregistrements")
        tables_existantes.append(table)
    except Exception as e:
        if "Could not find the table" in str(e):
            print(f"❌ {table}: n'existe pas")
        else:
            print(f"⚠️ {table}: {str(e)[:50]}")

# 3. ANALYSE DE LA TABLE EXPORTERS (si existe)
print("\n📋 2. ANALYSE DE LA TABLE PRINCIPALE")
print("-"*60)

if 'exporters' in tables_existantes:
    try:
        # Structure
        sample = supabase.table('exporters').select('*').limit(1).execute()
        if sample.data:
            colonnes = list(sample.data[0].keys())
            print(f"\n✅ Structure de 'exporters' ({len(colonnes)} colonnes):")
            for i, col in enumerate(sorted(colonnes)[:10], 1):  # 10 premières
                print(f"   {i:2d}. {col}")
            if len(colonnes) > 10:
                print(f"   ... et {len(colonnes)-10} autres")
        
        # Comptage par source
        total = supabase.table('exporters').select('*', count='exact').execute()
        print(f"\n📊 Total: {total.count} entreprises")
        
        if total.count > 0:
            sources = {}
            data = supabase.table('exporters').select('source_file').execute()
            for item in data.data:
                src = item.get('source_file')
                if src is None:
                    src = 'NULL'
                sources[src] = sources.get(src, 0) + 1
            
            print("\n📊 Répartition par source:")
            for src, count in sources.items():
                print(f"   • {src}: {count}")
                
    except Exception as e:
        print(f"❌ Erreur: {e}")
else:
    print("⚠️ Table 'exporters' non trouvée")

# 4. ANALYSE DES FICHIERS CSV
print("\n📋 3. ANALYSE DES FICHIERS CSV")
print("-"*60)

fichiers_csv = [
    ('data/wilayas_algerie.csv', 'wilayas'),
    ('data/secteurs_activite.csv', 'secteurs'),
    ('data/statuts_juridiques.csv', 'statuts')
]

for chemin, nom in fichiers_csv:
    print(f"\n📄 {nom}:")
    try:
        with open(chemin, 'r', encoding='utf-8') as f:
            reader = csv.reader(f)
            try:
                en_tete = next(reader)
                print(f"   ✅ Fichier trouvé")
                print(f"   📌 Colonnes ({len(en_tete)}): {', '.join(en_tete)}")
                
                # Compter les lignes
                lignes = sum(1 for _ in reader)
                print(f"   📊 Lignes de données: {lignes}")
                
            except StopIteration:
                print(f"   ⚠️ Fichier vide")
    except FileNotFoundError:
        print(f"   ❌ Fichier non trouvé")
    except Exception as e:
        print(f"   ❌ Erreur: {e}")

# 5. ANALYSE DU FICHIER ENTREPRISES
print("\n📋 4. ANALYSE DU FICHIER ENTREPRISES")
print("-"*60)

try:
    with open('data/private/entreprises_brutes.json', 'r', encoding='utf-8') as f:
        data = json.load(f)
        print(f"✅ Fichier trouvé: {len(data)} entreprises")
        
        if data and len(data) > 0:
            print(f"\n📌 Structure du premier élément:")
            for key, value in data[0].items():
                print(f"   • {key}: {type(value).__name__}")
                
except FileNotFoundError:
    print("❌ Fichier non trouvé: data/private/entreprises_brutes.json")
except Exception as e:
    print(f"❌ Erreur: {e}")

# 6. SYNTHÈSE ET RECOMMANDATIONS
print("\n" + "="*80)
print("📋 SYNTHÈSE EXPERT")
print("="*80)

print("\n🔴 PROBLÈMES IDENTIFIÉS:")
print("   1. Tables de référence non créées ou mal nommées")
print("   2. Fichiers CSV à vérifier (colonnes inconnues)")
print("   3. Structure de la base à clarifier")

print("\n🟢 PLAN D'ACTION RECOMMANDÉ:")
print("   ÉTAPE 1: Créer les tables de référence avec les bonnes colonnes")
print("   ÉTAPE 2: Adapter les scripts aux noms de colonnes réels")
print("   ÉTAPE 3: Importer les données")

print("\n" + "="*80)
print("✅ DIAGNOSTIC TERMINÉ")
print("="*80)